hdkeychain/extendedkey.go

```diff
@@ -195,6 +201,10 @@ func (k *ExtendedKey) Child(i uint32) (*ExtendedKey, error) {
 	// 3) Public extended key -> Non-hardened child public extended key
 	// 4) Public extended key -> Hardened child public extended key (INVALID!)
 
+	if k.depth == math.MaxUint8 {
```

Comment:
Rather than importing the entire `math` package for this sole constant, I would just prefer an unexported constant in the constants block near the top of the file.

```Go
maxUint8 = 1<<8 - 1
```

Then this would become:

```Go
if k.depth == maxUint8 {
```

The tests can still pull in the math package since they don't affect the primary package.